src/engine_core/assembly-test/StringExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using System.Runtime.InteropServices;
+
+internal static class StringExtensions {
+
+	/// <summary>
+	/// Extension method to convert a C# into a char* that can be used in C++ with ANSI encoding
+	/// </summary>
+	/// <param name="input">The string instance</param>
+	/// <returns>A char* in the global heap, encoded in ANSI</returns>
+	public static unsafe char* ToUnmanagedString(this string input)
+	{
+		return (char*)Marshal.StringToHGlobalAnsi(input);
```

Comment:
We should use `Marshal.StringToCoTaskMemUTF8` instead

Reply:
Got it, will address this tomorrow